core/src/test/java/org/apache/calcite/test/SqlValidatorTest.java

```diff
@@ -6289,6 +6296,35 @@ public boolean isBangEqualAllowed() {
     // Group by alias with strict conformance should fail.
     sql("select empno as e from emp group by ^e^")
         .withConformance(strict).fails("Column 'E' not found in any table");
+
+    sql("select floor(empno/2) as empno from emp group by floor(empno/2)")
+        .withConformance(strict).ok()
+        .withConformance(lenient).ok();
+  }
+
+  /**
+   * Tests validation of alias in function within GROUP BY.
+   *
+   * @see SqlConformance#isGroupByAlias()
+   */
+  @Test void testAliasInFunctionWithinGroupBy() {
+    final SqlConformanceEnum lenient = SqlConformanceEnum.LENIENT;
+    final SqlFunction date_add =
+        SqlBasicFunction.create(SqlKind.DATE_ADD, ReturnTypes.DATE,
+                OperandTypes.STRING_INTEGER)
+            .withFunctionType(SqlFunctionCategory.TIMEDATE);
+
+    sql("select date_add('2024-01-01', empno) as empno from emp group by ^year(empno)^")
```

Comment:
I don't really understand this test.

Reply:
now date_add function receives empno:integer as a parameter, returns a DATE type.
if empno in year(empno) is substituted by 'date_add('2024-01-01', empno)', the year function will not raise type inconsistency error.
but we expect it throws here, because the empno in year(empno) should not be substituted. the identifier in a function should not be substituted.
I add this test as @julianhyde has mentioned this case. I paste his comment here.

> Also I would like to see a SqlValidatorTest. Something like select date_add('2024-01-01', empno) as empno from emp group by year(empno), where there will be a type error (int vs date) if the wrong column is referenced.